app/code/Magento/Sitemap/Model/Sitemap.php

```diff
@@ -750,13 +736,9 @@ protected function _getStoreBaseDomain()
     }
 
     /**
-     * Get sitemap.xml URL according to all config options
-     *
-     * @param string $sitemapPath
-     * @param string $sitemapFileName
-     * @return string
+     * {@inheritdoc}
      */
-    public function getSitemapUrl($sitemapPath, $sitemapFileName)
+    public function getSitemapUrl($sitemapPath, $sitemapFileName): ?string
```

Comment:
We are not allowed to change public methods according to our [Backward Compatible Development Guide](https://devdocs.magento.com/guides/v2.2/contributor-guide/backward-compatible-development/)

Reply:
Removed getSitemapUrl from interface because it is not a clean dto method but has side effects.